DDF-1493 Remove gazetteer feature from search-ui-app feature.

dcruver opened the pull request:
Prior to this change, the webservice-gazetteer was being loaded as part of the
search-ui feature.  While the search-ui needs a gazetteer, the selection of a
specific gazetteer implementation should be left to configurations that are
loaded at a later time.  This change moves the gazetteer loading into the
install-profiles/*/features.xml.

@bdeining (Hero)
@kcwire 
@EIrwin

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf/177)
<!-- Reviewable:end -->

bdeining commented:
PR built successfully, verified that the webservice-gazetteer is still installed by default.  LGTM

kcwire commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[distribution/install-profiles/src/main/resources/features.xml, line 27 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/177#-JzRNsfXRGDh9CQ195si)** ([raw file](https://github.com/codice/ddf/blob/5f774bf74792cae3a16b8a51a0c0c8dd962ba191/distribution/install-profiles/src/main/resources/features.xml#L27)):</sup>
I believe only "apps" should be installed in the install profiles feature. I don't think this will actually do anything or at least this was not intended for individual features to be installed here.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

bdeining commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[distribution/install-profiles/src/main/resources/features.xml, line 27 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/177#-JzRNsfXRGDh9CQ195si)** ([raw file](https://github.com/codice/ddf/blob/5f774bf74792cae3a16b8a51a0c0c8dd962ba191/distribution/install-profiles/src/main/resources/features.xml#L27)):</sup>
If so, do we need this change to DDF at all?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

EIrwin commented:
<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java, line 91 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/177#-K--ROMaRQ4EueRJfOZW)** ([raw file](https://github.com/codice/ddf/blob/18ef8ef3951b466ada8d0ba2ba91cc746e6100aa/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java#L91)):</sup>
Did you mean to have this reference `geocoderServiceReference`? 

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

dcruver commented:
<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java, line 91 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/177#-K--ROMaRQ4EueRJfOZW)** ([raw file](https://github.com/codice/ddf/blob/18ef8ef3951b466ada8d0ba2ba91cc746e6100aa/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java#L91)):</sup>
Nope, it's an artifact from how this class evolved.  I'll update it

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

bdeining commented:
Review status: 2 of 14 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/util/impl/TestServiceFactoryImpl.java, line 46 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-F5phKc50_iyer1k8c)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/util/impl/TestServiceFactoryImpl.java#L46)):</sup>
Should be named setUp().  

---

<sup>**[catalog/spatial/geocoding/spatial-geocoding-geocoder/pom.xml, line 79 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-F6donJKNbvMwrw8Vm)** ([raw file](https://github.com/codice/ddf/blob/40e40117e599e0ed300506c6076cb207faefc46a/catalog/spatial/geocoding/spatial-geocoding-geocoder/pom.xml#L79)):</sup>
Should we remove the coverage threshold completely, or can it just be reduced?

---

<sup>**[catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/resources/OSGI-INF/blueprint/blueprint.xml, line 11 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-F74YlbeWCJFhSVjMo)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L11)):</sup>
Typo?

---

<sup>**[catalog/spatial/geocoding/spatial-geocoding-geocoder/src/test/java/org/codice/ddf/spatial/geocoder/endpoint/TestGeoCoderEndpoint.java, line 38 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-F7N6nkSKpU7QipDi5)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/spatial/geocoding/spatial-geocoding-geocoder/src/test/java/org/codice/ddf/spatial/geocoder/endpoint/TestGeoCoderEndpoint.java#L38)):</sup>
Should be setUp().

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

bdeining commented:
Reviewed 5 of 13 files at r2, 6 of 7 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

bdeining commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactory.java, line 23 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-FASbAc86ve5VqjCSJ)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactory.java#L23)):</sup>
There's an extra "be" in this sentence :sunglasses: 

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

bdeining commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

<sup>**[distribution/install-profiles/src/main/resources/features.xml, line 27 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/177#-JzRNsfXRGDh9CQ195si)** ([raw file](https://github.com/codice/ddf/blob/5f774bf74792cae3a16b8a51a0c0c8dd962ba191/distribution/install-profiles/src/main/resources/features.xml#L27)):</sup>
Now that the webservice-gazetteer has been moved into spatial-app, does this feature need to be in the install-profiles?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

bdeining commented:
PR builds successfully, webservice-gazetteer is installed by default.  :thumbsup: looks good.

---

Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

bdeining commented:
PR build was successful, web-gazetteer is installed by default.

dcruver commented:
Review status: 11 of 14 files reviewed at latest revision, 7 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactory.java, line 23 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-FASbAc86ve5VqjCSJ)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactory.java#L23)):</sup>
removed.
<!-- Reviewable comment -K-JmV54T7qQNfXmXOxM:1675637715 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/util/impl/TestServiceFactoryImpl.java, line 46 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-F5phKc50_iyer1k8c)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/util/impl/TestServiceFactoryImpl.java#L46)):</sup>
fixed.
<!-- Reviewable comment -K-JmSNyzqUUphnwnTOS:-1620474753 -->

---

<sup>**[catalog/spatial/geocoding/spatial-geocoding-geocoder/pom.xml, line 79 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-F6donJKNbvMwrw8Vm)** ([raw file](https://github.com/codice/ddf/blob/40e40117e599e0ed300506c6076cb207faefc46a/catalog/spatial/geocoding/spatial-geocoding-geocoder/pom.xml#L79)):</sup>
Once the actual coverage exceeds our standard, we remove the threshold completely.  That way, the threshold is only listed in pom files for deficient packages. 
<!-- Reviewable comment -K-JlmrJ2mRaXHlibzPC:2056701625 -->

---

<sup>**[catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/resources/OSGI-INF/blueprint/blueprint.xml, line 11 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-F74YlbeWCJFhSVjMo)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L11)):</sup>
removed.
<!-- Reviewable comment -K-JlkbCieZFh_GAQEb2:1675637715 -->

---

<sup>**[catalog/spatial/geocoding/spatial-geocoding-geocoder/src/test/java/org/codice/ddf/spatial/geocoder/endpoint/TestGeoCoderEndpoint.java, line 38 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-F7N6nkSKpU7QipDi5)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/spatial/geocoding/spatial-geocoding-geocoder/src/test/java/org/codice/ddf/spatial/geocoder/endpoint/TestGeoCoderEndpoint.java#L38)):</sup>
fixed.
<!-- Reviewable comment -K-JlhpFJy0rlPbXDUpu:-1620474753 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

dcruver commented:
Review status: 10 of 13 files reviewed at latest revision, 7 unresolved discussions, some commit checks pending.

---

<sup>**[distribution/install-profiles/src/main/resources/features.xml, line 27 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/177#-JzRNsfXRGDh9CQ195si)** ([raw file](https://github.com/codice/ddf/blob/5f774bf74792cae3a16b8a51a0c0c8dd962ba191/distribution/install-profiles/src/main/resources/features.xml#L27)):</sup>
removed.
<!-- Reviewable comment -K-Jp-dXoaTwzlgR8csk:1675637715 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

kcwire commented:
Reviewed 5 of 13 files at r2, 5 of 7 files at r3, 1 of 1 files at r4.
Review status: 10 of 13 files reviewed at latest revision, 14 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactory.java, line 28 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-Jm-S7bNKbLqlCEfnU)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactory.java#L28)):</sup>
This is example doesn't seem correct.
<!-- Reviewable comment -K-Jm-S7bNKbLqlCEfnV:504169949 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactory.java, line 39 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-Jl9iZh4gMl_hySr8F)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactory.java#L39)):</sup>
This can be a little confusing since there already exists a "ManagedServiceFactory" which has a completely different meaning.  Maybe we can think up another name?
<!-- Reviewable comment -K-Jl9iZh4gMl_hySr8G:-37042994 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java, line 47 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-JlWX3eSGnG-CX6V_b)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java#L47)):</sup>
Why XLogger? We Typically just do `LoggerFactory.getLogger(clazz)`.
<!-- Reviewable comment -K-JlWX3eSGnG-CX6V_c:-95692085 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java, line 112 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-JmOJEqQnxDsx6WeZp)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java#L112)):</sup>
Clean up the comment
<!-- Reviewable comment -K-JmOJEqQnxDsx6WeZq:1737072391 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java, line 113 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-JmVFN7XfWGxk5vl73)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java#L113)):</sup>
potential NPE from `getBundleContext()` & `getService()` 

https://osgi.org/javadoc/r4v43/core/org/osgi/framework/BundleContext.html#getService(org.osgi.framework.ServiceReference)
<!-- Reviewable comment -K-JmVFN7XfWGxk5vl74:1179983759 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/util/impl/TestServiceFactoryImpl.java, line 66 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-Jn-X4Wa9l-DFc_Qyi)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/util/impl/TestServiceFactoryImpl.java#L66)):</sup>
assertThat provides a very clean output when tests do fail and is preferred. Also add these as static imports.
<!-- Reviewable comment -K-Jn-X4Wa9l-DFc_Qyj:97118558 -->

---

<sup>**[catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/endpoint/GeoCoderFactory.java, line 21 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-JnwagkYnjTf1F1I9M)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/endpoint/GeoCoderFactory.java#L21)):</sup>
I feel like this defeats the purpose of the ServiceFactory interface if I have to extend it every time I want to use it.  Ideally I would just create and instance of the ServiceFactoryImpl.  We may not even need/want the interface.
<!-- Reviewable comment -K-JnwagkYnjTf1F1I9N:-1380109418 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

dcruver commented:
Review status: 5 of 13 files reviewed at latest revision, 8 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/endpoint/GeoCoderFactory.java, line 21 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-JnwagkYnjTf1F1I9M)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/endpoint/GeoCoderFactory.java#L21)):</sup>
I made several changes to address this - 

1) renamed ServiceFactoryImpl to ServiceSelector
2) removed the ServiceFactory interface
3) added the ServiceSelectorConverter
4) removed the GeoCoderServiceFactory GeoCoderServiceFactoryImpl
5) changed the blueprint file to wire up the ServiceSelector directly
<!-- Reviewable comment -K-Jz5iG3TNhpu3aAf4Y:-1654504009 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

dcruver commented:
Review status: 5 of 13 files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java, line 47 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-JlWX3eSGnG-CX6V_b)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java#L47)):</sup>
Because other classes (SortedServiceList, ListConverter, etc.) in that package are also using XLogger.  This was patterned on that example.  
<!-- Reviewable comment -K-PgqqIzWghgj9llB3u:127743147 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java, line 112 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-JmOJEqQnxDsx6WeZp)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java#L112)):</sup>
changed.
<!-- Reviewable comment -K-PlfcCXWUddB0qayQR:-674089985 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java, line 113 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-JmVFN7XfWGxk5vl73)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java#L113)):</sup>
changed.
<!-- Reviewable comment -K-PlhfDQwyXlTTTIJtW:-674089985 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/util/impl/TestServiceFactoryImpl.java, line 66 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-Jn-X4Wa9l-DFc_Qyi)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/util/impl/TestServiceFactoryImpl.java#L66)):</sup>
changed.
<!-- Reviewable comment -K-PrzyPxTOMFaKMUqUo:-674089985 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

bdeining commented:
Review status: 5 of 13 files reviewed at latest revision, 10 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/FirstElementServiceSelectionStrategy.java, line 28 \[r7\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-PpPPJLNz06eRAjV9-)** ([raw file](https://github.com/codice/ddf/blob/1e9e2faec66ce3d24e4c0611d919a6c206440285/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/FirstElementServiceSelectionStrategy.java#L28)):</sup>
{@link class} might be better for this, instead of @see
<!-- Reviewable comment -K-PpPPJLNz06eRAjV90:-733857702 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java, line 72 \[r7\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-Pq2Z5f8rFnWwssnqP)** ([raw file](https://github.com/codice/ddf/blob/1e9e2faec66ce3d24e4c0611d919a6c206440285/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java#L72)):</sup>
{@link class}
<!-- Reviewable comment -K-Pq2Z6_1fkBO1oXXax:-1892441151 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java, line 84 \[r7\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-PoeIcPNobK37BVlFu)** ([raw file](https://github.com/codice/ddf/blob/1e9e2faec66ce3d24e4c0611d919a6c206440285/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java#L84)):</sup>
private static final Logger LOGGER = LoggerFactory.getLogger(ServiceSelector.class);
<!-- Reviewable comment -K-PoeIdhkC_bxkiWw0g:-655875811 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java, line 168 \[r7\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-PqNgoVotfbwS67TGM)** ([raw file](https://github.com/codice/ddf/blob/1e9e2faec66ce3d24e4c0611d919a6c206440285/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java#L168)):</sup>
unbindService
<!-- Reviewable comment -K-PqNgoVotfbwS67TGN:1067151002 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

bdeining commented:
Reviewed 8 of 13 files at r6, 1 of 1 files at r7, 1 of 1 files at r8, 1 of 1 files at r9.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

dcruver commented:
Review status: 11 of 13 files reviewed at latest revision, 10 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/FirstElementServiceSelectionStrategy.java, line 28 \[r7\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-PpPPJLNz06eRAjV9-)** ([raw file](https://github.com/codice/ddf/blob/1e9e2faec66ce3d24e4c0611d919a6c206440285/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/FirstElementServiceSelectionStrategy.java#L28)):</sup>
changed to @link
<!-- Reviewable comment -K-QDkeqWZQfmo0hsBjL:1485528390 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java, line 72 \[r7\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-Pq2Z5f8rFnWwssnqP)** ([raw file](https://github.com/codice/ddf/blob/1e9e2faec66ce3d24e4c0611d919a6c206440285/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java#L72)):</sup>
changed to @link
<!-- Reviewable comment -K-QDnG1HBxLRyprQQXs:1485528390 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java, line 84 \[r7\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-PoeIcPNobK37BVlFu)** ([raw file](https://github.com/codice/ddf/blob/1e9e2faec66ce3d24e4c0611d919a6c206440285/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java#L84)):</sup>
changed as suggested.
<!-- Reviewable comment -K-QDotnUg1yvNm9T8eR:1993362888 -->

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java, line 168 \[r7\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-PqNgoVotfbwS67TGM)** ([raw file](https://github.com/codice/ddf/blob/1e9e2faec66ce3d24e4c0611d919a6c206440285/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceSelector.java#L168)):</sup>
fixed.
<!-- Reviewable comment -K-QDw6ghu5SQzs7LDTd:-1620474753 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

bdeining commented:
Reviewed 2 of 2 files at r10.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->

dcruver commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---

<sup>**[catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java, line 47 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/177#-K-JlWX3eSGnG-CX6V_b)** ([raw file](https://github.com/codice/ddf/blob/8953b1690722e44855aa4ff9343cf4c218ca96f9/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ServiceFactoryImpl.java#L47)):</sup>
Changed this to use the regular log4j Logger.  I'll create a story to remove investigate and possibly remove XLogger from the remainder of DDF.
<!-- Reviewable comment -K-QFRf1jyMoxcMTUPcm:-2130363089 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/177)
<!-- Sent from Reviewable.io -->
